### PR TITLE
Add/sitehealth enqueued assets

### PR DESCRIPTION
**Pull request by manuelRod**
This was the original plugin to be moved to modules:
https://github.com/audrasjb/site-health-audit-enqueued-assets/blob/master/site-health-audit-enqueued-assets.php

### CODE REVIEW

File: modules/audit-enqueued-assets/load.php [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+/**
+ * Gets total of enqueued scripts.
+ *
+ * @since 1.0.0
+ *
+ * @return int|false Number of total scripts or false if transient hasn't been set.
+ */
+function aea_get_total_enqueued_scripts() {
+	$enqueued_scripts = false;
+	if ( get_transient( 'aea_enqueued_scripts' ) ) {
```
Comment (pbearne) [inline modules/audit-enqueued-assets/load.php]: can we call get_transient once

File: modules/audit-enqueued-assets/load.php [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+/**
+ * Gets total of enqueued styles.
+ *
+ * @since 1.0.0
+ *
+ * @return int|false Number of total styles or false if transient hasn't been set.
+ */
+function aea_get_total_enqueued_styles() {
+	$enqueued_styles = false;
+	if ( get_transient( 'aea_enqueued_styles' ) ) {
```
Comment (pbearne) [inline modules/audit-enqueued-assets/load.php]: can we call get_transient once

File: modules/audit-enqueued-assets/load.php [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+		global $wp_scripts;
+		$enqueued_scripts = array();
+
+		foreach ( $wp_scripts->queue as $handle ) {
+			$src = $wp_scripts->registered[ $handle ]->src;
+			if ( $src && ! strpos( $src, 'wp-includes' ) ) {
+				$enqueued_scripts[] = $src;
+			}
+		}
+		set_transient( 'aea_enqueued_scripts', $enqueued_scripts, 12 * HOUR_IN_SECONDS );
```
Comment (pbearne) [inline modules/audit-enqueued-assets/load.php]: love to see a filter here for TTL maybe

And should we be invaliding the cache for theme change and plugin install

File: modules/audit-enqueued-assets/load.php [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => esc_html__( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			esc_html__( 'The amount of enqueued CSS assets is acceptable.', 'performance-lab' )
+		),
+		'actions'     => '',
```
Comment (pbearne) [inline modules/audit-enqueued-assets/load.php]: in all the actions blocks
Can we add a button/link to reset the count
I can see that someone could be trying to improve the score but it not changing as the transient set to 1 day :-)

**Review by pbearne — CHANGES_REQUESTED** [review]
The tests look really good well done
Most of the comments are around the invaliding the cache

**Comment by manuelRod** [comment]
This is the WP issue: https://github.com/WordPress/performance/issues/25

**Comment by manuelRod** [comment]
@pbearne ready for the second code review, I added everything we talked about.

File: modules/audit-enqueued-assets/load.php [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+/**
+ * Invalidate both transients/cache on user clean_aea_audit action.
+ *
+ * @since 1.0.0
+ */
+function clean_aea_audit_action() {
+	if ( isset( $_GET['action'] ) && 'clean_aea_audit' === $_GET['action'] ) {
+		check_admin_referer( 'clean_aea_audit' );
+		invalidate_cache_transients();
```
Comment (pbearne) [inline modules/audit-enqueued-assets/load.php]: can/should we do a redirect hare to clear the querystring
wp_safe_redirect( remove_query_arg( 'action', 'clean_aea_audit' ) );

**Review by pbearne — CHANGES_REQUESTED** [review]
I don't like leaving a query string with an action in the URL as it reruns the action on a page reload

File: modules/audit-enqueued-assets/load.php [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+ */
+
+/**
+ * Audit enqueued scripts in the frontend. Ignore /wp-includes scripts.
+ *
+ * It will save information in a transient for 12 hours.
+ *
+ * @since 1.0.0
+ */
+function aea_audit_enqueued_scripts() {
```
Comment (ThierryA) [inline modules/audit-enqueued-assets/load.php]: All prefix needs to be updated to `perflab_`

File: modules/audit-enqueued-assets/load.php [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			esc_html__( 'The amount of enqueued CSS assets is acceptable.', 'performance-lab' )
+		),
+		'actions'     => '',
+		'test'        => 'enqueued_css_assets',
+	);
+
+	if ( $enqueued_styles > 10 ) {
```
Comment (ThierryA) [inline modules/audit-enqueued-assets/load.php]: This signal needs to be more robust. The number of stylesheets is somehow a week signal, what really matter is the totally size. We need to grab and sum the size of all stylesheets and define a threshold for the warning.

File: modules/audit-enqueued-assets/load.php [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			esc_html__( 'The amount of enqueued JS assets is acceptable.', 'performance-lab' )
+		),
+		'actions'     => '',
+		'test'        => 'enqueued_js_assets',
+	);
+
+	if ( $enqueued_scripts > 10 ) {
```
Comment (ThierryA) [inline modules/audit-enqueued-assets/load.php]: Same comment as for the stylesheet, we need to grab and sum the total size of assets loaded.

**Review by ThierryA — CHANGES_REQUESTED** [review]
Thanks for working on this 😄   I believe this is still WIP although the this PR is not appearing as a "Draft". Nonetheless I left a few early comments without doing a detailed review yet. The approach to trigger an asset warning is quite weak at the moment and need some more work.

By the way, why opening a PR agains this fork rather than on the https://github.com/WordPress/performance repo, that makes it hard for others to follow along with the review. Could we continue the conversation on the main repo please?

File: modules/audit-enqueued-assets/load.php (hunk 5 above)
Comment (manuelRod) [reply modules/audit-enqueued-assets/load.php]: you are right! fixed.

File: modules/audit-enqueued-assets/load.php (hunk 7 above)
Comment (manuelRod) [reply modules/audit-enqueued-assets/load.php]: @ThierryA I've opened the PR  in WP https://github.com/WordPress/performance/pull/54 we can follow up in there.

Comment (ThierryA) [reply modules/audit-enqueued-assets/load.php]: Thank you @manuelRod, much appreciated. Left the same comments on the new PR to continue the conversation there.

**Pull request merged**